github/repos.go

```diff
@@ -736,6 +739,9 @@ type ProtectionRequest struct {
 	RequiredPullRequestReviews *PullRequestReviewsEnforcementRequest `json:"required_pull_request_reviews"`
 	EnforceAdmins              bool                                  `json:"enforce_admins"`
 	Restrictions               *BranchRestrictionsRequest            `json:"restrictions"`
+	RequiredLinearHistory      bool                                  `json:"required_linear_history"`
```

Comment:
Unfortunately, this won't work either.  Because these are optional fields, these three need to be type `*bool` and not `bool`.
They also need the `,omiteempty` flag on them